conan/tools/microsoft/msbuild.py

```diff
@@ -48,11 +47,75 @@ def command(self, sln, targets=None):
 
         return cmd
 
-    def build(self, sln, targets=None):
-        cmd = self.command(sln, targets=targets)
+    def build(self, sln, targets=None, force_import_generated_files=False):
+        cmd = self.command(sln, targets=targets, force_import_generated_files=force_import_generated_files)
         self._conanfile.run(cmd)
 
     @staticmethod
     def get_version(_):
         return NotImplementedError("get_version() method is not supported in MSBuild "
                                    "toolchain helper")
+
+    def _get_concrete_props_file(self, root_props_file):
```

Comment:
I am not sure why the specific file needs to be passed, if the general toolchain file already contains conditionals to pick the right configuration one.

Reply:
> if the general toolchain file already contains conditionals to pick the right configuration one.

Because it doesn't work.
I'm testing with several CCI recipes for the moment, and properties are ignored when they come only from props file (msbuild complains about PlatformToolset since it's not overridden, and when there is a hardcoded WholeProgramOptimization I can't disable it through a custom property in MSBuildToolchain). When I extract these values from toolchain file and inject them in command line, it works.

Reply:
(obviously, I would be more than happy if a smart solution not involving to parse xml files could be found)

Reply:
I still see a lot of complexity in this functionality.
Wouldn't it be way more straightforward to just inject the ``<import>`` section in the ``vcxproj`` directly?